src/sys/socket/addr.rs

```diff
@@ -64,6 +64,24 @@ impl InetAddr {
         }
     }
 
+    /// Creates an `IpAddr` struct from libc's sockaddr.
+    ///
+    /// Note that this supports only IPv4 and IPv6 address families.
+    pub unsafe fn from_libc_sockaddr(addr: *mut libc::sockaddr) -> Option<InetAddr> {
+        if addr.is_null() {
+            None
+        } else {
+            let family = (*addr).sa_family as u32;
+            match mem::transmute::<u32, AddressFamily>(family) {
```

Comment:
Instead of using transmute here, it's better to provide an `AddressFamily::from_u32` method.